anvio/drivers/sourmash.py

```diff
@@ -54,46 +62,272 @@ def check_program(self):
         old_wd = os.getcwd()
         os.chdir(input_path)
-        os.mkdir('output')
+        if not os.path.exists('output'):
+            os.mkdir('output')
+        else:
+            pass
 
         self.progress.new('Sourmash')
-        self.progress.update('Computing fasta signatures...')
+        self.progress.update('Computing fasta signatures for kmer=%d, scale=%d' % (self.kmer_size, self.scale))
 
         scale = '--scaled=%i' % self.scale
         compute_command = [self.program_name, 'compute',
                            '-k', self.kmer_size,
                            '-f', scale]
         compute_command.extend(fasta_files)
 
-        exit_code = utils.run_command(compute_command, self.log_file_path)
+        exit_code = utils.run_command(compute_command, self.log_file_path, remove_log_file_if_exists=False)
         if int(exit_code):
             self.progress.end()
             raise ConfigError("sourmash returned with non-zero exit code, there may be some errors.\
                               Please check the log file `%s` for details. Offending command: \
                               `%s` ..." % (self.log_file_path, ' '.join([str(x) for x in compute_command[:7]])))
 
-        self.progress.update('Computing distance matrix...')
+        self.progress.update('Computing similarity matrix for kmer=%d, scale=%d' % (self.kmer_size, self.scale))
         compare_command = [self.program_name, 'compare',
                            '-k', self.kmer_size,
-                           '--csv', os.path.join('output', 'mash_distance.txt')]
+                           '--csv', os.path.join('output', report_name + '.txt')]
         for f in fasta_files:
             compare_command.append(f + ".sig")
 
-        exit_code = utils.run_command(compare_command, self.log_file_path)
+        exit_code = utils.run_command(compare_command, self.log_file_path, remove_log_file_if_exists=False)
         if int(exit_code):
             self.progress.end()
             raise ConfigError("sourmash returned with non-zero exit code, there may be some errors.\
                               Please check the log file `%s` for details. Offending command: \
                               `%s` ..." % (self.log_file_path, ' '.join([str(x) for x in compute_command[:7]])))
 
-        matrix = utils.get_TAB_delimited_file_as_dictionary(os.path.join('output', 'mash_distance.txt'), indexing_field=-1, separator=',')
+        self.results[report_name] = utils.get_TAB_delimited_file_as_dictionary(os.path.join('output', report_name + '.txt'),
+                                                                               indexing_field=-1,
+                                                                               separator=',')
 
         self.progress.end()
 
         # restore old working directory
         os.chdir(old_wd)
 
-        return matrix
+        return self.results
+
+
+class IterateKmerSourmash(Sourmash):
```

Comment:
should we keep this class regardless?

Reply:
No lol

Reply:
I'm going to remove it tomorrow and then merge. Does that sound good?

Reply:
It does :)